pages/docs/reference/collection-grouping.md

```diff
@@ -0,0 +1,54 @@
+---
+type: doc
+layout: reference
+category: "Collections"
+title: "Grouping"
+---
+
+# Grouping
+
+The Kotlin standard library provides extension functions for grouping collection elements.
+The basic function `groupBy()` takes a lambda function and returns a `Map`.
+In this map, each key is the lambda result and the corresponding value is the `List` of elements on which this result is returned.
+This function can be used, for example, to group a list of `String`s by their first letter. 
+
+You can also call `groupBy()` with a second lambda argument – a value transformation function.
+In the result map of `groupBy()` with two lambdas, the keys produced by `keySelector` function are mapped to the results of the value transformation function instead of the original elements.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.3">
+
+```kotlin
+fun main() {
+//sampleStart
+    val numbers = listOf("one", "two", "three", "four", "five")
+
+    println(numbers.groupBy { it.first().toUpperCase() })
+    println(numbers.groupBy(keySelector = { it.first() }, valueTransform = { it.toUpperCase() }))
+//sampleEnd
+}
+```
+</div>
+
+If you want to group elements and then apply an operation to all groups at one time, use the function `groupingBy()`.
+It returns an instance of the `Grouping` type.
+The `Grouping` instance stores all the returned groups together and lets you easily apply operations to them all.
```

Comment:
`Grouping` doesn't store all groups. That's the whole point of it, otherwise `groupBy` would be enough. It is lazy as sequences, i.e. it only stores the group key selector and the source of elements.

Reply:
Fixed the description.